### DISCUSSION TITLE
why do cvxpy code using cvxpy.logistic say DCP Error ?

### DISCUSSION THREAD

**Discussion by Hideki105** [cvxpy/cvxpy#1985] [Q&A]
Why cvxpy said DCP ERROR ?
```
import cvxpy as cp

-cp.logistic(-yi*xi.T@beta ) <= s1[i]
```
cvxpy.error.DCPError: Problem does not follow DCP rules. Specifically: The following constraints are not DCP: -logistic([[ 0.34119207 -1.09868391]] @ var315) <= var318[0] , because the following subexpressions are not: |-- -logistic([[ 0.34119207 -1.09868391]] @ var315) <= var318[0] -logistic([[ 1.17921346 -0.94992175]] @ var315) <= var318[1] , because the following subexpressions are not: |-- -logistic([[ 1.17921346 -0.94992175]] @ var315) <= var318[1] -logistic([[ 1.55135968 -0.84252651]] @ var315) <= var318[2] , because the following subexpressions are not: |-- -logistic([[ 1.55135968 -0.84252651]] @ var315) <= var318[2]

I can not understand why cvxpy tell me DCP ERROR.

**Comment by phschiele** [comment]
@Hideki105 Are both `yi` and `xi` variables? Multiplying variables is not DCP.
Further, `cp.logistic()` is convex, so `-cp.logistic()` is concave. 
Only constraints of the form `convex <= concave` are DCP (i.e., concave terms must be on the "greater" side).
